pkg/kubectl/cmd/cp.go

```diff
@@ -244,22 +297,46 @@ func untarAll(reader io.Reader, destFile, prefix string) error {
 			}
 			break
 		}
+
+		mode := header.FileInfo().Mode()
+		if isFirstFile {
+			if !header.FileInfo().IsDir() && strings.HasSuffix(prefix, "/") {
```

Comment:
while this is a great improvement over the current behavior, maybe we should consider removing the trailing slash from the filename in this case without returning the error.

cc @fabianofranz

Reply:
thanks for the reply.
so you mean just copy the existing file without returning the error.
I refer the behaviour of cp a file from host to container.It will return an error with hint 
error:  not a directory.
so if your suggestion is better choice for cp command, I think it is necessary to keep cp command  in same pace  with both directions

Reply:
@juanvallejo  are you still there? so it  supposed to return a error or just copy the file without any error?
which one is better?

Reply:
Sorry about the delay.
At least with tools like `scp`, I'm able to specify a filename with a trailing slash, and have the command still work as expected.

I'd prefer copying the file without error, regardless of the trailing slash

Reply:
OK, I will modify this and rebase this patch after your patch #51215 merged